Running-Mastodon/Development-guide.md

```diff
@@ -61,6 +61,43 @@ You can check code quality with:
 
     rubocop
 
+## OpenBSD
+
+Follow the Linux setup as described above, but with these considerations:
+
+- If you use a Ruby version manager (chruby, rbenv, rvm, etc.), you _must_
+  configure Ruby with `CC=clang CXX=clang++`. This instructs Ruby to use that
+  compiler when compiling native C gems.
+- Many native C gems need to be told about `/usr/local`. You can do this by
+  configuring a `build.gem_name` value using `bundle config`.
+- Any C gem that uses mkmf.rb's `pkg_config` method might fail if the linker
+  produces warnings, as happens when a library links with `sprintf(3)`. The
+  `cld3` gem uses `pkg_config('protobuf')`; if you have protobuf installed but
+  it cannot be found while building the gem, this is likely the problem. You
+  will need to directly modify `mkmf.rb` to get this to install.
+
+The bundle configuration as of Mastodon 2.0's Gemfile:
+
+```sh
+bundle config build.nokogiri --use-system-libraries --with-xml2-include=/usr/local/include/libxml2/ --with-opt-include=/usr/local/include --with-xslt-include=/usr/local/include/libxslt --with-exslt-include=/usr/local/include/libexslt --with-xml2-lib=/usr/local/lib
+bundle config build.charlock_holms --with-icu-dir=-I/usr/local --with-opt-dir=/usr/local
+bundle config build.idn-ruby --with-idn-dir=/usr/local
+```
+
+Modify `mfmk.rb`:
```

Comment:
where is this file? is this really a BSD specific issue? why don't other platforms run into this problem?